package Nursultan;

import minecraft.class01830;
import minecraft.class03875;
import minecraft.class03877;
import minecraft.class03894;
import minecraft.class03909;
import minecraft.class03912;
import minecraft.class07321;

public class class09530 implements class01830, class03894 {
   private final class03877 N;
   private long y = class07321.L;
   private double L;

   public class09530(class03877 var1) {
      this.N = var1;
   }

   public class03909 i() {
      return class03909.field_36564;
   }

   public class03877 u() {
      return this.N;
   }

   public double N(class03875 var1) {
      int var2 = var1.y();
      int var3 = var1.u();
      long var4 = class07321.u(var2, var3);
      if (this.y == var4) {
         return this.L;
      } else {
         this.y = var4;
         double var6 = this.N.N(var1);
         this.L = var6;
         return var6;
      }
   }

   public void N(double[] var1, class03912 var2) {
      this.N.N(var1, var2);
   }
}
